pkg/providers/v1/aws.go

```diff
@@ -1289,11 +1295,27 @@ func init() {
 		var creds *credentials.Credentials
 		if cfg.Global.RoleARN != "" {
 			klog.Infof("Using AWS assumed role %v", cfg.Global.RoleARN)
+			stsClient := sts.New(sess)
```

Comment:
`init()` is getting a bit long, you might want to push the sts client generation code into a helper function?